### ISSUE TITLE
What about chars like ÄÖÜßàéè in Alnum/Alpha?

### ISSUE THREAD

**Issue by ghost** [Respect/Validation#627]
Hello!
I've some trouble with validating some user input using the framework. The problem is, that some languages allow chars like **Ä Ö Ü ß à è é ú** but the Alnum and Alpha validators do not allow them. Any idea? Or is there any validator available that allows all available letters?

Regards and thanks for your great work!

**Comment by alganet** [comment]
Unfortunately, we don't have rules to validate such chars yet. 

A previous discussion about this started on issue #92 and it was suggested that the native `setlocale()` function might affect the behavior of the `ctype_alpha` and `ctype_alnum` functions (that we use as the basis for these rules), so calling `setlocale(LC_ALL, 'es_ES'); (changing es_ES to the appropriate language) might work. I never used it this way though.

**Comment by alganet** [comment]
Reading #92 again I've remembered the Charset rule. Might be handy: http://respect.github.io/Validation/docs/charset.html

**Comment by 7rust** [comment]
Alright! In my opinion Respect needs at least one Validator that allows all or the most common letters (Latin + normal letters). This rule might could use the ASCII table and just don't allow any chars that aren't letters.

**Comment by pultho** [comment]
What about a rule like `alphaLatin($pAdditional= "");`

Taking a look at [the following UTF basic latin table](http://www.utf8-zeichentabelle.de/unicode-utf8-table.pl?number=1024&utf8=dec) we can see that all latin chars are between:



* **65 - 90**
* **97 - 122**
* **195 - 207** (most latin special chars)

Now we just need to check whether **each char** is within this range by using `ord("Ä");`.

This wouldn't be a perfect solution but a step forward on supporting latin chars (which is definitely required in most European countries).

Maybe something like:
```
   public function validate($input)
   {
        foreach(str_split($input) as $c)
        {
            if(!(
                (in_array(ord($c), range(65, 90)))
                || (in_array(ord($c), range(97, 122)))
                || (in_array(ord($c), range(195, 207))) )
            )return false;
        }
        return true;
   }

```

**Comment by henriquemoody** [comment]
I created #762 in order to fix this issue.

However, before I merge it I would like to understand in which scenario do you intent to use this validator?

**Comment by pultho** [comment]
I think it's helpful for validating texts or names so that you can make sure that your input doesn't contain any special characters or symbols.

**Comment by henriquemoody** [comment]
Now is more clear what is the intention here.

According to [Wikipedia](https://en.wikipedia.org/wiki/Alphanumeric):
> Alphanumeric is a combination of alphabetic and numeric characters, and is used to describe the collection of Latin letters and Arabic digits or a text constructed from this collection.

That said, the one we have is correct, and changing it would make it wrong. What is needed is a new rule, with a new name and a new purpose. I'm planning to finish it this week, and I'm gonna make it more clear on the documentation. The rule will be probably called _Name_ but I'm very open for suggestions.

**Comment by nickl-** [comment]
Closed: as per #762 Won't fix
